site/src/pages/WorkspacesPage/BatchActions.tsx

```diff
@@ -0,0 +1,142 @@
+import TextField from "@mui/material/TextField";
+import { Box } from "@mui/system";
+import { deleteWorkspace, startWorkspace, stopWorkspace } from "api/api";
+import { Workspace } from "api/typesGenerated";
+import { ConfirmDialog } from "components/Dialogs/ConfirmDialog/ConfirmDialog";
+import { displayError } from "components/GlobalSnackbar/utils";
+import { useState } from "react";
+import { useMutation } from "react-query";
+import { MONOSPACE_FONT_FAMILY } from "theme/constants";
+
+export const useBatchActions = (options: {
+  onSuccess: () => Promise<void>;
+}) => {
+  const { onSuccess } = options;
+
+  const startAllMutation = useMutation({
+    mutationFn: async (workspaces: Workspace[]) => {
+      return Promise.all(
+        workspaces.map((w) =>
+          startWorkspace(w.id, w.latest_build.template_version_id),
+        ),
+      );
+    },
+    onSuccess,
+    onError: () => {
+      displayError("Failed to start workspaces");
+    },
+  });
+
+  const stopAllMutation = useMutation({
+    mutationFn: async (workspaces: Workspace[]) => {
+      return Promise.all(workspaces.map((w) => stopWorkspace(w.id)));
+    },
+    onSuccess,
+    onError: () => {
+      displayError("Failed to stop workspaces");
+    },
+  });
+
+  const deleteAllMutation = useMutation({
+    mutationFn: async (workspaces: Workspace[]) => {
+      return Promise.all(workspaces.map((w) => deleteWorkspace(w.id)));
+    },
+    onSuccess,
+    onError: () => {
+      displayError("Failed to delete workspaces");
+    },
+  });
+
+  return {
+    startAll: startAllMutation.mutateAsync,
+    stopAll: stopAllMutation.mutateAsync,
+    deleteAll: deleteAllMutation.mutateAsync,
+    isLoading:
+      startAllMutation.isLoading ||
+      stopAllMutation.isLoading ||
+      deleteAllMutation.isLoading,
+  };
+};
+
+type BatchDeleteConfirmationProps = {
+  checkedWorkspaces: Workspace[];
+  open: boolean;
+  isLoading: boolean;
+  onClose: () => void;
+  onConfirm: () => void;
+};
+
+export const BatchDeleteConfirmation = (
+  props: BatchDeleteConfirmationProps,
+) => {
+  const { checkedWorkspaces, open, onClose, onConfirm, isLoading } = props;
+  const [confirmation, setConfirmation] = useState({ value: "", error: false });
```

Comment:
What values is `value` expected to have right now? So far, it looks like `""` or `DELETE`?
I'm wondering if `value` could have its own type, if we're only going to be supporting a discrete set of operations